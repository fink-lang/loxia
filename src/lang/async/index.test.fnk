{fink2js} = import '../../testing'
{describe, it, expect, to_match_snapshot} = import '@fink/jest'


describe 'await', fn:
  it 'compiles', fn:

    expect
      fink2js `
        task1 = fn foo: - await foo

        task2 = fn foo: await (foo + 4)

        task3 = fn foo:
          bar = await foo ()
          spam = await foo _
          bar + 123

        a_gen = unfold curr=0:
          match shrub:
            spam: await ni curr
            else: curr + 1

        await ni
      `
      to_match_snapshot



