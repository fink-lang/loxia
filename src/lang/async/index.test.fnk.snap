// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`await compiles 1`] = `
"export const task1 = async (foo) => -(await foo);
export const task2 = async (foo) => await (foo + 4);
export const task3 = async (foo) => {
  const bar = await foo();
  const spam = await foo();
  return bar + 123;
};
export const a_gen = async function* unfold(ˆinitial_2) {
  let ˆaccu_3 = 0;

  while (true) {
    const curr = ˆaccu_3;

    let _do_result;

    ˆmatch_5: {
      const ˆvalue_4 = shrub;

      if (ˆvalue_4 === spam) {
        _do_result = await ni(curr);
        break ˆmatch_5;
      }

      {
        _do_result = curr + 1;
        break ˆmatch_5;
      }
    }

    const ˆresult_1 = _do_result;
    _do_result = undefined;
    yield ˆresult_1;
    ˆaccu_3 = ˆresult_1;
  }
};
await ni;"
`;
